pkg/controller/deployment/deployment_controller.go

```diff
@@ -415,6 +416,15 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		glog.V(4).Infof("Ignoring paused deployment %s/%s", d.Namespace, d.Name)
 		return nil
 	}
+	if d.Spec.RollbackTo != nil {
+		revision := d.Spec.RollbackTo.Revision
+		if err := dc.rollback(d, &revision); err != nil {
```

Comment:
Yes there will be an update conflict. The reason that this didn't cause a problem is because after the conflict happens it'll return the error and then do the new rc revision update at the next round `syncDeployment`. I'll return here to avoid this conflict.